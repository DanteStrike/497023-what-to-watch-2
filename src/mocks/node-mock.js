export const createNodeMock = (element) => {
  if (element.type === `video`) {
    return {
      src: null,
      isMuted: null,
      poster: null,
      addEventListener: () => {}
    };
  }
  return null;
};
