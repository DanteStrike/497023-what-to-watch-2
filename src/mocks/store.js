import StoreNameSpace from "../reducers/store-name-space.js";
import {comments} from "./comments.js";
import {normalizedFilmsData} from "./films";

export const initStore = {
  [StoreNameSpace.APP]: {
    isReady: false,
    videoPlayerFilmID: -1
  },

  [StoreNameSpace.FILMS]: {
    data: {
      allIDs: [],
      byIDs: {}
    },

    promo: {
      filmID: null
    }
  },

  [StoreNameSpace.COMMENTS]: {
    postCommentState: {
      isSuccess: false,
      error: {
        isError: false,
        msg: ``
      }
    },
    curFilmComments: []
  },

  [StoreNameSpace.GENRES]: {
    currentFilter: `All genre`,
    data: {
      genres: [`All genre`],
      byGenres: {
        "All genre": [],
      }
    }
  },

  [StoreNameSpace.CATALOG]: {
    displayedFilmsAmount: 0
  },

  [StoreNameSpace.USER]: {
    auth: {
      isAuth: false,
      error: {
        isError: false,
        target: ``,
        msg: ``
      }
    },

    toggleFavoriteStatus: {
      isSuccess: false,
      error: {
        isError: false
      }
    },

    data: {
      id: null,
      email: ``,
      name: ``,
      avatarUrl: ``,
      myListFilmsIDs: []
    }
  }
};

export const loadedStore = {
  [StoreNameSpace.APP]: {
    isReady: true,
    videoPlayerFilmID: 1
  },

  [StoreNameSpace.FILMS]: {
    data: normalizedFilmsData,

    promo: {
      filmID: 3
    }
  },

  [StoreNameSpace.COMMENTS]: {
    postCommentState: {
      isSuccess: true,
      error: {
        isError: false,
        msg: ``
      }
    },
    curFilmComments: comments
  },

  [StoreNameSpace.GENRES]: {
    currentFilter: `All genre`,
    data: {
      genres: [`All genre`, `genreOne`, `genreTwo`],
      byGenres: {
        "All genre": [1, 3],
        "genreOne": [1],
        "genreTwo": [3]
      }
    }
  },

  [StoreNameSpace.CATALOG]: {
    displayedFilmsAmount: 0
  },

  [StoreNameSpace.USER]: {
    auth: {
      isAuth: true,
      error: {
        isError: true,
        target: `target`,
        msg: `any`
      }
    },

    toggleFavoriteStatus: {
      isSuccess: true,
      error: {
        isError: false
      }
    },

    data: {
      id: 1,
      email: `Oliver.conner@gmail.com`,
      name: `Oliver.conner`,
      avatarUrl: `img/1.png`,
      myListFilmsIDs: [3]
    }
  },
};
