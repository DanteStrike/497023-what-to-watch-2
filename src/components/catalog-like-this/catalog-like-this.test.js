import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter as Router} from 'react-router-dom';

import {Provider} from "react-redux";
import {createStore} from "redux";

import {initStore} from "../../mocks/store.js";
import {createNodeMock} from "../../mocks/node-mock.js";
import CatalogLikeThis from "../catalog-like-this/catalog-like-this.jsx";


it(`Render correctly CatalogLikeThis component`, () => {
  const store = createStore(() => initStore);
  const component = renderer
    .create(
        <Provider store={store}>
          <Router>
            <CatalogLikeThis
              id={0}
              genre={`any`}
            />
          </Router>
        </Provider>,
        {
          createNodeMock
        }
    ).toJSON();

  expect(component).toMatchSnapshot();
});
