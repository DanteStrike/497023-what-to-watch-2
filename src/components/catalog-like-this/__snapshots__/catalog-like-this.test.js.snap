// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly CatalogLikeThis component 1`] = `
<section
  className="catalog catalog--like-this"
>
  <h2
    className="catalog__title"
  >
    More like this
  </h2>
  <div
    className="catalog__movies-list"
  />
</section>
`;
